### PR TITLE
Fixes #26324 - Add API for policies ENC

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

**Comment by ares** [comment]
I'm sorry, rubocop..

```
Inspecting 109 files
........................................C....................................................................

Offenses:

/home/jenkins/workspace/foreman_openscap-pull-request/database/postgresql/label/fast/ruby/2.5/plugin/app/models/concerns/foreman_openscap/smart_proxy_extensions.rb:9:20: C: Style/MutableConstant: Freeze mutable objects assigned to constants.
      PORT_MATCH = /:(\d+)\z/
                   ^^^^^^^^^^
```

**Comment by xprazak2** [comment]
Fixing rubocop, there is https://github.com/theforeman/hammer_cli_foreman_openscap/pull/23, but output formatting will need some adjustments.

**Comment by ares** [comment]
needs rebase now

**Comment by ares** [comment]
Otherwise works well, except for one nit, I need to specify host id as number, it does not accept friendly id. Not sure if this is a problem or not. I also found n+1 issue we have, but is out of the scope for this PR, I'll send separate PR.

**Comment by xprazak2** [comment]
I rebased and changed the id to `:dotted_identifier`

File: config/routes.rb [hunk 1]
```diff
@@ -103,4 +103,17 @@
       post 'openscap_proxy_changed'
     end
   end
+
+  namespace :api do
+    scope "(:api_version)", :module => :v2,
+                            :defaults => { :api_version => 'v2' },
+                            :api_version => /v2/,
+                            :constraints => ApiConstraints.new(:version => 2, :default => true) do
+      resources :hosts, :only => [] do
```
Comment (ares) [inline config/routes.rb]: this needs to be wrapped in `constraints(:id => /[^\/]+/) do ... end `

Comment (xprazak2) [reply config/routes.rb]: Updated with constraints.

**Comment by ares** [comment]
APJ

**Comment by ares** [comment]
Thanks @xprazak2, merging

**Pull request merged**